### PR TITLE
Issue 20915 generate secure password

### PR DESCRIPTION
**Pull request by alfredo-dotcms**
### Proposed Changes
https://user-images.githubusercontent.com/751424/132049638-83cf022e-8c7b-4e71-a529-ec621175ecce.mp4

### Checklist
- [ ] Tests
- [ ] Translations
- [ ] Security Implications Contemplated (add notes if applicable)

### Additional Info
** any additional useful context or info **

### Screenshots
Original             |  Updated
:-------------------------:|:-------------------------:
** original screenshot **  |  ** updated screenshot **

### CODE REVIEW

File: apps/dotcms-ui/src/app/api/services/dot-generate-secure-password/dot-generate-secure-password.service.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import { Injectable } from '@angular/core';
+import { Observable, Subject } from 'rxjs';
+
+@Injectable()
+export class DotGenerateSecurePasswordService {
+    private data: Subject<{ [key: string]: any }> = new Subject<{ [key: string]: any }>();
```
Comment (fmontes) [inline apps/dotcms-ui/src/app/api/services/dot-generate-secure-password/dot-generate-secure-password.service.ts:6]: Can we type `any`?

File: apps/dotcms-ui/src/app/view/components/_common/dot-generate-secure-password/dot-generate-secure-password.module.ts [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+@NgModule({
+    declarations: [DotGenerateSecurePasswordComponent],
+    exports: [DotGenerateSecurePasswordComponent],
+    providers: [DotGenerateSecurePasswordService, DotClipboardUtil],
+    imports: [
+        ButtonModule,
+        CommonModule,
+        DotDialogModule,
+        DotPipesModule
```
Comment (fmontes) [inline apps/dotcms-ui/src/app/view/components/_common/dot-generate-secure-password/dot-generate-secure-password.module.ts:18]: Do we need `DotPipesModule`? I thought we had it global.

**Review by fmontes — CHANGES_REQUESTED** [review]

File: apps/dotcms-ui/src/app/api/services/dot-generate-secure-password/dot-generate-secure-password.service.ts [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+
+    get showDialog$(): Observable<{ [key: string]: any }> {
+        return this.data.asObservable();
+    }
+
+    /**
+     * Notify subscribers with new data
+     *
+     * @param string data
```
Comment (hmoreras) [inline apps/dotcms-ui/src/app/api/services/dot-generate-secure-password/dot-generate-secure-password.service.ts:15]: fix type in doc

**Review by hmoreras — CHANGES_REQUESTED** [review]

File: apps/dotcms-ui/src/app/view/components/_common/dot-generate-secure-password/dot-generate-secure-password.module.ts (hunk 2 above)
Comment (alfredo-dotcms) [reply apps/dotcms-ui/src/app/view/components/_common/dot-generate-secure-password/dot-generate-secure-password.module.ts:18]: yes it's needed

**Review by fmontes — DISMISSED** [review]

**Pull request merged**